File: tests/test_modularity_audit.py
import os
import sys
from pathlib import Path


def test_server_modularity_grep():
    """Verify that server.py does NOT contain logic that should be modularized."""
    server_path = Path("zena_mode/server.py")
    if not server_path.exists():
        return  # Skip if called from wrong CWD

    with open(server_path, "r", encoding="utf-8") as f:
        content = f.read()

    # These strings should NO LONGER be in server.py as primary logic
    monolithic_strings = [
        "model_manager.get_popular_models()",
        "urllib.parse.parse_qs",
        "sd.rec(",
        "wav.write(",
        "mimetypes.guess_type(file_path)",
    ]

    for s in monolithic_strings:
        assert s not in content, f"Monolithic logic '{s}' still found in server.py! Should be in handlers."


def test_handler_registration():
    """Verify that handlers are correctly imported in server.py."""
    from zena_mode.server import ZenAIOrchestrator

    # Check inheritance
    from zena_mode.handlers.base import BaseZenHandler

    assert issubclass(ZenAIOrchestrator, BaseZenHandler)

    # Check delegated calls (we can check if ModelHandler is used in do_GET)
    import inspect

    source = inspect.getsource(ZenAIOrchestrator.do_GET)
    assert "ModelHandler.handle_get(self)" in source
    assert "VoiceHandler.handle_get(self)" in source
    assert "StaticHandler.handle_get(self)" in source


def test_vulnerability_scan_grep():
    """Tougher test: grep for hardcoded secrets or unsafe binds."""
    # Grep all python files for hardcoded localhost IPs in config strings
    # (Except where intentional like in server bind)
    pass

File: rust_output/src/test_modularity_audit.rs
use anyhow::{Result, Context};
use std::fs::File;
use std::io::{self, Read, Write};
use std::path::PathBuf;

/// Verify that server::py does NOT contain logic that should be modularized.
pub fn test_server_modularity_grep() -> Result<()> {
    // Verify that server::py does NOT contain logic that should be modularized.
    let mut server_path = PathBuf::from("zena_mode/server::py".to_string());
    if !server_path.exists() {
        return;
    }
    let mut f = File::open(server_path)?;
    {
        let mut content = f.read();
    }
    let mut monolithic_strings = vec!["model_manager::get_popular_models()".to_string(), "urllib::parse.parse_qs".to_string(), "sd.rec(".to_string(), "wav.write(".to_string(), "mimetypes.guess_type(file_path)".to_string()];
    for s in monolithic_strings.iter() {
        assert!(!content.contains(&s), "Monolithic logic '{}' still found in server::py! Should be in handlers.", s);
    }
}

/// Verify that handlers are correctly imported in server::py.
pub fn test_handler_registration() -> () {
    // Verify that handlers are correctly imported in server::py.
    // TODO: from zena_mode.server import ZenAIOrchestrator
    // TODO: from zena_mode.handlers::base import BaseZenHandler
    assert!(issubclass(ZenAIOrchestrator, BaseZenHandler));
    // TODO: import inspect
    let mut source = inspect::getsource(ZenAIOrchestrator.do_GET);
    assert!(source.contains(&"ModelHandler.handle_get(self)".to_string()));
    assert!(source.contains(&"VoiceHandler.handle_get(self)".to_string()));
    assert!(source.contains(&"StaticHandler.handle_get(self)".to_string()));
}

/// Tougher test: grep for hardcoded secrets or unsafe binds.
pub fn test_vulnerability_scan_grep() -> () {
    // Tougher test: grep for hardcoded secrets or unsafe binds.
    // pass
}
